packages/vue-language-core/src/utils/directorySharedTypes.ts

```diff
@@ -17,13 +17,17 @@ type __VLS_Prettify<T> = {
 	[K in keyof T]: T[K];
 } & {};
 
+// @ts-ignore
+type __VLS_VueGlobalComponents = import('vue').GlobalComponents;
+// @ts-ignore
+type __VLS_VueRuntimeCoreGlobalComponents = import('@vue/runtime-core').GlobalComponents;
+// @ts-ignore
+type __VLS_VueRuntimeDomGlobalComponents = import('@vue/runtime-dom').GlobalComponents;
+
 type __VLS_GlobalComponents =
-	// @ts-ignore
-	__VLS_PickNotAny<import('vue').GlobalComponents, {}>
-	// @ts-ignore
-	& __VLS_PickNotAny<import('@vue/runtime-core').GlobalComponents, {}>
-	// @ts-ignore
-	& __VLS_PickNotAny<import('@vue/runtime-dom').GlobalComponents, {}>
+	& __VLS_PickNotAny<__VLS_VueGlobalComponents & __VLS_VueGlobalComponents, {}>
```

Comment:
Maybe include a comment explaining this "hack"?
Personally I don't understand what's the purpose of doing `&` against same type.

Reply:
Done!

Reply:
You wrote that it's a hack but I still don't understand what it actually does. What difference does it make to types?

Reply:
Well I added a link in the pr description